vsphere-iso crashes

When filing a bug please include the following headings, if possible. 

<!--
Any example text in this template can be deleted.
-->

#### Overview of the Issue

Please provide a clear and concise description of the issue you are experiencing.

vsphere-iso crashes

#### Reproduction Steps

Please provide the steps to reproduce the issue.

Run `packer init` and then `packer build` on the provided build file

### Packer Version

Please provide the Packer version.

Packer v1.9.1

Also verified on Packer v1.7.5

### Plugin Version and Builders

Please provide the plugin version.

1.2.0

Please select the builder.

- [x] `vsphere-iso`
- [ ] `vsphere-clone`

### VMware vSphere Version

Please provide the VMware vSphere version.

N/A, crashes before connecting to vSphere

### Guest Operating System

N/A, crashes before this is relevant

### Simplified Packer Buildfile

If the file is longer than a few dozen lines, please include the URL to the [gist](https://gist.github.com/) of the log or use the [GitHub detailed format](https://gist.github.com/ericclemmons/b146fe5da72ca1f706b2ef72a20ac39d) instead of posting it directly in the issue.

```packer
packer {
  required_plugins {
    vsphere = {
      version = "= v1.2.0"
      source  = "github.com/hashicorp/vsphere"
    }
  }
}

source "vsphere-iso" "linux-sles" {
  vcenter_server = "foo"
  username       = "foo"
  password       = "foo"
  cluster        = "foo"
  vm_name        = "foo"
  ssh_username   = "foo"
  storage {
    disk_size    = 40960
  }
}

build {
  sources = ["source.vsphere-iso.linux-sles"]
}
```

### Operating System and Environment Details

Please add any information you can provide about the environment.

<!--
    Example:
     - Operating System: macOS Big Sur (Intel)
-->

### Log Fragments and `crash.log` Files

Include appropriate log fragments. If the log is longer than a few dozen lines, please include the URL to the [gist](https://gist.github.com/) of the log or use the [Github detailed format](https://gist.github.com/ericclemmons/b146fe5da72ca1f706b2ef72a20ac39d) instead of posting it directly in the issue.

Set the env var `PACKER_LOG=1` for maximum log detail.

Build command was `PACKER_LOG=1 PACKER_CONFIG_DIR=./.packer.d packer build . |& tee out.txt`

<details>
<summary>crash.log</summary>

```text
2023/06/15 09:11:55 [INFO] Packer version: 1.9.1 [go1.20.4 linux amd64]
2023/06/15 09:11:55 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:55 [TRACE] discovering plugins in /home/nfaulkne/bin
2023/06/15 09:11:55 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:55 [TRACE] discovering plugins in .packer.d/.packer.d/plugins
2023/06/15 09:11:55 [DEBUG] Discovered plugin: vsphere = /home/nfaulkne/test_packer/.packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64
2023/06/15 09:11:55 [DEBUG] Discovered plugin: vsphere = /home/nfaulkne/test_packer/.packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64
2023/06/15 09:11:55 [INFO] found external [clone iso supervisor] builders from vsphere plugin
2023/06/15 09:11:55 [INFO] found external [-packer-default-plugin-name- template] post-processors from vsphere plugin
2023/06/15 09:11:55 [TRACE] discovering plugins in .
2023/06/15 09:11:55 [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/06/15 09:11:55 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:55 [INFO] PACKER_CONFIG env var set; attempting to open config file: .packer.d/.packerconfig
2023/06/15 09:11:55 [WARN] Config file doesn't exist: .packer.d/.packerconfig
2023/06/15 09:11:55 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:55 [INFO] Setting cache directory: /home/nfaulkne/.cache/packer
2023/06/15 09:11:55 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:55 [TRACE] listing potential installations for "github.com/hashicorp/vsphere" that match "= v1.2.0". plugingetter.ListInstallationsOptions{FromFolders:[]string{"/home/nfaulkne/bin/packer", ".", ".packer.d/.packer.d/plugins"}, BinaryInstallationOptions:plugingetter.BinaryInstallationOptions{APIVersionMajor:"5", APIVersionMinor:"0", OS:"linux", ARCH:"amd64", Ext:"", Checksummers:[]plugingetter.Checksummer{plugingetter.Checksummer{Type:"sha256", Hash:(*sha256.digest)(0xc0004eab80)}}}}
2023/06/15 09:11:55 [TRACE] version "v1.1.2" of file ".packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64" does not match constraint "= v1.2.0"
2023/06/15 09:11:55 [TRACE] Found the following "github.com/hashicorp/vsphere" installations: [{.packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 v1.2.0}]
2023/06/15 09:11:55 [INFO] found external [clone iso supervisor] builders from vsphere plugin
2023/06/15 09:11:55 [INFO] found external [-packer-default-plugin-name- template] post-processors from vsphere plugin
2023/06/15 09:11:55 [TRACE] Starting external plugin .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 start builder iso
2023/06/15 09:11:55 Starting plugin: .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 []string{".packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64", "start", "builder", "iso"}
2023/06/15 09:11:55 Waiting for RPC address for: .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64
2023/06/15 09:11:55 Received unix RPC address for .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64: addr is /tmp/packer-plugin3600082970
2023/06/15 09:11:55 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 2023/06/15 09:11:55 Plugin address: unix /tmp/packer-plugin3600082970
2023/06/15 09:11:55 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 2023/06/15 09:11:55 Waiting for connection...
2023/06/15 09:11:55 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 2023/06/15 09:11:55 Serving a plugin connection...
2023/06/15 09:11:55 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 2023/06/15 09:11:55 [TRACE] starting builder iso
2023/06/15 09:11:55 ui: [1;32mvsphere-iso.linux-sles: output will be in this color.[0m
2023/06/15 09:11:55 ui: 
2023/06/15 09:11:55 Build debug mode: false
2023/06/15 09:11:55 Force build: false
2023/06/15 09:11:55 On error: 
2023/06/15 09:11:55 Waiting on builds to complete...
2023/06/15 09:11:55 Starting build run: vsphere-iso.linux-sles
2023/06/15 09:11:55 Running builder: 
2023/06/15 09:11:55 [INFO] (telemetry) Starting builder vsphere-iso.linux-sles
2023/06/15 09:12:00 ui: [0;32m    vsphere-iso.linux-sles: Closing sessions ....[0m
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: panic: interface conversion: interface is nil, not driver.Driver
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: goroutine 54 [running]:
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/common.(*StepConnect).Cleanup(0xc0006a22a0?, {0x29a3d68, 0xc0006a2270})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/common/step_connect.go:71 +0x9d
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-sdk/multistep.(*BasicRunner).Run(0xc0006a2390, {0x29a3898, 0xc0006c60c0}, {0x29a3d68, 0xc0006a2270})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	github.com/hashicorp/packer-plugin-sdk@v0.4.0/multistep/basic_runner.go:85 +0x23a
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/iso.(*Builder).Run(0xc0004bb000, {0x29a3898, 0xc0006c60c0}, {0x29a6488?, 0xc0006a2240}, {0x29846a0?, 0xc0006ae020})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/iso/builder.go:164 +0x15c3
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-sdk/rpc.(*BuilderServer).Run(0xc00040c000, 0x708080?, 0xc000708010)
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	github.com/hashicorp/packer-plugin-sdk@v0.4.0/rpc/builder.go:120 +0x1e3
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: reflect.Value.call({0xc00070c480?, 0xc00000eb50?, 0x13?}, {0x2451a45, 0x4}, {0xc000130ef8, 0x3, 0x3?})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	reflect/value.go:556 +0x845
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: reflect.Value.Call({0xc00070c480?, 0xc00000eb50?, 0x3?}, {0xc000a94ef8, 0x3, 0x3})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	reflect/value.go:339 +0xbf
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: net/rpc.(*service).call(0xc00040c040, 0xc000933d4a?, 0x0?, 0xc00040e5f0, 0xc00019e280, 0x609225?, {0x1f55520?, 0xc00070800c?, 0x6091a6?}, {0x1ed4f40, ...}, ...)
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	net/rpc/server.go:381 +0x226
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: created by net/rpc.(*Server).ServeCodec
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	net/rpc/server.go:478 +0x3fe
2023/06/15 09:12:00 [INFO] (telemetry) ending vsphere-iso.linux-sles
2023/06/15 09:12:00 .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64: plugin process exited
2023/06/15 09:12:00 ui error: [1;31mBuild 'vsphere-iso.linux-sles' errored after 4 seconds 451 milliseconds: unexpected EOF[0m
2023/06/15 09:12:00 ui: 
==> Wait completed after 4 seconds 451 milliseconds
2023/06/15 09:12:00 machine readable: error-count []string{"1"}
2023/06/15 09:12:00 ui error: 
==> Some builds didn't complete successfully and had errors:
2023/06/15 09:12:00 machine readable: vsphere-iso.linux-sles,error []string{"unexpected EOF"}
2023/06/15 09:12:00 ui error: --> vsphere-iso.linux-sles: unexpected EOF
2023/06/15 09:12:00 ui: 
==> Builds finished but no artifacts were created.
2023/06/15 09:12:00 [INFO] (telemetry) Finalizing.
2023/06/15 09:12:00 waiting for all plugin processes to complete...

```

</details>
<details>
<summary>packer build output</summary>

```text
2023/06/15 09:11:55 [INFO] Packer version: 1.9.1 [go1.20.4 linux amd64]
2023/06/15 09:11:55 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:55 [TRACE] discovering plugins in /home/nfaulkne/bin
2023/06/15 09:11:55 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:55 [TRACE] discovering plugins in .packer.d/.packer.d/plugins
2023/06/15 09:11:55 [DEBUG] Discovered plugin: vsphere = /home/nfaulkne/test_packer/.packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64
2023/06/15 09:11:55 [DEBUG] Discovered plugin: vsphere = /home/nfaulkne/test_packer/.packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64
2023/06/15 09:11:55 [INFO] found external [clone iso supervisor] builders from vsphere plugin
2023/06/15 09:11:55 [INFO] found external [-packer-default-plugin-name- template] post-processors from vsphere plugin
2023/06/15 09:11:55 [TRACE] discovering plugins in .
2023/06/15 09:11:55 [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/06/15 09:11:55 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:55 [INFO] PACKER_CONFIG env var set; attempting to open config file: .packer.d/.packerconfig
2023/06/15 09:11:55 [WARN] Config file doesn't exist: .packer.d/.packerconfig
2023/06/15 09:11:55 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:55 [INFO] Setting cache directory: /home/nfaulkne/.cache/packer
2023/06/15 09:11:55 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:55 [TRACE] listing potential installations for "github.com/hashicorp/vsphere" that match "= v1.2.0". plugingetter.ListInstallationsOptions{FromFolders:[]string{"/home/nfaulkne/bin/packer", ".", ".packer.d/.packer.d/plugins"}, BinaryInstallationOptions:plugingetter.BinaryInstallationOptions{APIVersionMajor:"5", APIVersionMinor:"0", OS:"linux", ARCH:"amd64", Ext:"", Checksummers:[]plugingetter.Checksummer{plugingetter.Checksummer{Type:"sha256", Hash:(*sha256.digest)(0xc0004eab80)}}}}
2023/06/15 09:11:55 [TRACE] version "v1.1.2" of file ".packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64" does not match constraint "= v1.2.0"
2023/06/15 09:11:55 [TRACE] Found the following "github.com/hashicorp/vsphere" installations: [{.packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 v1.2.0}]
2023/06/15 09:11:55 [INFO] found external [clone iso supervisor] builders from vsphere plugin
2023/06/15 09:11:55 [INFO] found external [-packer-default-plugin-name- template] post-processors from vsphere plugin
2023/06/15 09:11:55 [TRACE] Starting external plugin .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 start builder iso
2023/06/15 09:11:55 Starting plugin: .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 []string{".packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64", "start", "builder", "iso"}
2023/06/15 09:11:55 Waiting for RPC address for: .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64
2023/06/15 09:11:55 Received unix RPC address for .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64: addr is /tmp/packer-plugin3600082970
2023/06/15 09:11:55 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 2023/06/15 09:11:55 Plugin address: unix /tmp/packer-plugin3600082970
2023/06/15 09:11:55 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 2023/06/15 09:11:55 Waiting for connection...
2023/06/15 09:11:55 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 2023/06/15 09:11:55 Serving a plugin connection...
2023/06/15 09:11:55 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 2023/06/15 09:11:55 [TRACE] starting builder iso
2023/06/15 09:11:55 Build debug mode: false
[1;32mvsphere-iso.linux-sles: output will be in this color.[0m

2023/06/15 09:11:55 Force build: false
2023/06/15 09:11:55 On error: 
2023/06/15 09:11:55 Waiting on builds to complete...
2023/06/15 09:11:55 Starting build run: vsphere-iso.linux-sles
2023/06/15 09:11:55 Running builder: 
2023/06/15 09:11:55 [INFO] (telemetry) Starting builder vsphere-iso.linux-sles
[0;32m    vsphere-iso.linux-sles: Closing sessions ....[0m
[1;31mBuild 'vsphere-iso.linux-sles' errored after 4 seconds 451 milliseconds: unexpected EOF[0m

==> Wait completed after 4 seconds 451 milliseconds

==> Some builds didn't complete successfully and had errors:
--> vsphere-iso.linux-sles: unexpected EOF

==> Builds finished but no artifacts were created.
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: panic: interface conversion: interface is nil, not driver.Driver
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: goroutine 54 [running]:
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/common.(*StepConnect).Cleanup(0xc0006a22a0?, {0x29a3d68, 0xc0006a2270})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/common/step_connect.go:71 +0x9d
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-sdk/multistep.(*BasicRunner).Run(0xc0006a2390, {0x29a3898, 0xc0006c60c0}, {0x29a3d68, 0xc0006a2270})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	github.com/hashicorp/packer-plugin-sdk@v0.4.0/multistep/basic_runner.go:85 +0x23a
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/iso.(*Builder).Run(0xc0004bb000, {0x29a3898, 0xc0006c60c0}, {0x29a6488?, 0xc0006a2240}, {0x29846a0?, 0xc0006ae020})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/iso/builder.go:164 +0x15c3
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-sdk/rpc.(*BuilderServer).Run(0xc00040c000, 0x708080?, 0xc000708010)
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	github.com/hashicorp/packer-plugin-sdk@v0.4.0/rpc/builder.go:120 +0x1e3
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: reflect.Value.call({0xc00070c480?, 0xc00000eb50?, 0x13?}, {0x2451a45, 0x4}, {0xc000130ef8, 0x3, 0x3?})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	reflect/value.go:556 +0x845
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: reflect.Value.Call({0xc00070c480?, 0xc00000eb50?, 0x3?}, {0xc000a94ef8, 0x3, 0x3})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	reflect/value.go:339 +0xbf
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: net/rpc.(*service).call(0xc00040c040, 0xc000933d4a?, 0x0?, 0xc00040e5f0, 0xc00019e280, 0x609225?, {0x1f55520?, 0xc00070800c?, 0x6091a6?}, {0x1ed4f40, ...}, ...)
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	net/rpc/server.go:381 +0x226
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: created by net/rpc.(*Server).ServeCodec
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	net/rpc/server.go:478 +0x3fe
2023/06/15 09:12:00 [INFO] (telemetry) ending vsphere-iso.linux-sles
2023/06/15 09:12:00 .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64: plugin process exited
==> Wait completed after 4 seconds 451 milliseconds
2023/06/15 09:12:00 machine readable: error-count []string{"1"}
panic: interface conversion: interface is nil, not driver.Driver
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: goroutine 54 [running]:
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/common.(*StepConnect).Cleanup(0xc0006a22a0?, {0x29a3d68, 0xc0006a2270})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/common/step_connect.go:71 +0x9d
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-sdk/multistep.(*BasicRunner).Run(0xc0006a2390, {0x29a3898, 0xc0006c60c0}, {0x29a3d68, 0xc0006a2270})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	github.com/hashicorp/packer-plugin-sdk@v0.4.0/multistep/basic_runner.go:85 +0x23a
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/iso.(*Builder).Run(0xc0004bb000, {0x29a3898, 0xc0006c60c0}, {0x29a6488?, 0xc0006a2240}, {0x29846a0?, 0xc0006ae020})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	github.com/hashicorp/packer-plugin-vsphere/builder/vsphere/iso/builder.go:164 +0x15c3
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: github.com/hashicorp/packer-plugin-sdk/rpc.(*BuilderServer).Run(0xc00040c000, 0x708080?, 0xc000708010)
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	github.com/hashicorp/packer-plugin-sdk@v0.4.0/rpc/builder.go:120 +0x1e3
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: reflect.Value.call({0xc00070c480?, 0xc00000eb50?, 0x13?}, {0x2451a45, 0x4}, {0xc000130ef8, 0x3, 0x3?})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	reflect/value.go:556 +0x845
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: reflect.Value.Call({0xc00070c480?, 0xc00000eb50?, 0x3?}, {0xc000a94ef8, 0x3, 0x3})
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	reflect/value.go:339 +0xbf
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: net/rpc.(*service).call(0xc00040c040, 0xc000933d4a?, 0x0?, 0xc00040e5f0, 0xc00019e280, 0x609225?, {0x1f55520?, 0xc00070800c?, 0x6091a6?}, {0x1ed4f40, ...}, ...)
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	net/rpc/server.go:381 +0x226
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: created by net/rpc.(*Server).ServeCodec
2023/06/15 09:12:00 packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64 plugin: 	net/rpc/server.go:478 +0x3fe
2023/06/15 09:12:00 [INFO] (telemetry) ending vsphere-iso.linux-sles
2023/06/15 09:12:00 .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64: plugin process exited
2023/06/15 09:12:00 ui error: [1;31mBuild 'vsphere-iso.linux-sles' errored after 4 seconds 451 milliseconds: unexpected EOF[0m
2023/06/15 09:12:00 ui: 
==> Wait completed after 4 seconds 451 milliseconds
2023/06/15 09:12:00 machine readable: error-count []string{"1"}
2023/06/15 09:12:00 ui error: 
==> Some builds didn't complete successfully and had errors:
2023/06/15 09:12:00 machine readable: vsphere-iso.linux-sles,error []string{"unexpected EOF"}
2023/06/15 09:12:00 ui error: --> vsphere-iso.linux-sles: unexpected EOF
2023/06/15 09:12:00 ui: 
==> Builds finished but no artifacts were created.
2023/06/15 09:12:00 [INFO] (telemetry) Finalizing.
2023/06/15 09:12:00 waiting for all plugin processes to complete...

==> Some builds didn't complete successfully and had errors:
2023/06/15 09:12:00 machine readable: vsphere-iso.linux-sles,error []string{"unexpected EOF"}
==> Builds finished but no artifacts were created.
2023/06/15 09:12:00 [INFO] (telemetry) Finalizing.
2023/06/15 09:12:00 waiting for all plugin processes to complete...


!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Packer crashed! This is always indicative of a bug within Packer.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Packer[1] so that we can fix this.

[1]: https://github.com/hashicorp/packer/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

```

</details>
<details>
<summary>packer build output on plugin version 1.1.2</summary>

```text
2023/06/15 09:11:33 [INFO] Packer version: 1.9.1 [go1.20.4 linux amd64]
2023/06/15 09:11:33 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:33 [TRACE] discovering plugins in /home/nfaulkne/bin
2023/06/15 09:11:33 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:33 [TRACE] discovering plugins in .packer.d/.packer.d/plugins
2023/06/15 09:11:33 [DEBUG] Discovered plugin: vsphere = /home/nfaulkne/test_packer/.packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64
2023/06/15 09:11:33 [DEBUG] Discovered plugin: vsphere = /home/nfaulkne/test_packer/.packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64
2023/06/15 09:11:33 [INFO] found external [clone iso supervisor] builders from vsphere plugin
2023/06/15 09:11:33 [INFO] found external [-packer-default-plugin-name- template] post-processors from vsphere plugin
2023/06/15 09:11:33 [TRACE] discovering plugins in .
2023/06/15 09:11:33 [INFO] PACKER_CONFIG env var not set; checking the default config file path
2023/06/15 09:11:33 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:33 [INFO] PACKER_CONFIG env var set; attempting to open config file: .packer.d/.packerconfig
2023/06/15 09:11:33 [WARN] Config file doesn't exist: .packer.d/.packerconfig
2023/06/15 09:11:33 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:33 [INFO] Setting cache directory: /home/nfaulkne/.cache/packer
2023/06/15 09:11:33 Detected config directory from env var: ./.packer.d
2023/06/15 09:11:33 [TRACE] listing potential installations for "github.com/hashicorp/vsphere" that match "= v1.1.2". plugingetter.ListInstallationsOptions{FromFolders:[]string{"/home/nfaulkne/bin/packer", ".", ".packer.d/.packer.d/plugins"}, BinaryInstallationOptions:plugingetter.BinaryInstallationOptions{APIVersionMajor:"5", APIVersionMinor:"0", OS:"linux", ARCH:"amd64", Ext:"", Checksummers:[]plugingetter.Checksummer{plugingetter.Checksummer{Type:"sha256", Hash:(*sha256.digest)(0xc0004ea000)}}}}
2023/06/15 09:11:33 [TRACE] version "v1.2.0" of file ".packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.2.0_x5.0_linux_amd64" does not match constraint "= v1.1.2"
2023/06/15 09:11:33 [TRACE] Found the following "github.com/hashicorp/vsphere" installations: [{.packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64 v1.1.2}]
2023/06/15 09:11:33 [INFO] found external [clone iso supervisor] builders from vsphere plugin
2023/06/15 09:11:33 [INFO] found external [-packer-default-plugin-name- template] post-processors from vsphere plugin
2023/06/15 09:11:33 [TRACE] Starting external plugin .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64 start builder iso
2023/06/15 09:11:33 Starting plugin: .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64 []string{".packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64", "start", "builder", "iso"}
2023/06/15 09:11:33 Waiting for RPC address for: .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64
2023/06/15 09:11:33 Received unix RPC address for .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64: addr is /tmp/packer-plugin2176889207
2023/06/15 09:11:33 packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64 plugin: 2023/06/15 09:11:33 Plugin address: unix /tmp/packer-plugin2176889207
2023/06/15 09:11:33 packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64 plugin: 2023/06/15 09:11:33 Waiting for connection...
2023/06/15 09:11:33 packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64 plugin: 2023/06/15 09:11:33 Serving a plugin connection...
2023/06/15 09:11:33 packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64 plugin: 2023/06/15 09:11:33 [TRACE] starting builder iso
2023/06/15 09:11:33 Build debug mode: false
2023/06/15 09:11:33 Force build: false
2023/06/15 09:11:33 On error: 
2023/06/15 09:11:33 Waiting on builds to complete...
2023/06/15 09:11:33 Starting build run: vsphere-iso.linux-sles
2023/06/15 09:11:33 Running builder: 
2023/06/15 09:11:33 [INFO] (telemetry) Starting builder vsphere-iso.linux-sles
[1;32mvsphere-iso.linux-sles: output will be in this color.[0m

2023/06/15 09:11:38 [INFO] (telemetry) ending vsphere-iso.linux-sles
==> Wait completed after 4 seconds 524 milliseconds
2023/06/15 09:11:38 machine readable: error-count []string{"1"}
==> Some builds didn't complete successfully and had errors:
2023/06/15 09:11:38 machine readable: vsphere-iso.linux-sles,error []string{"Post \"https://foo/sdk\": dial tcp: lookup foo on 172.29.176.1:53: server misbehaving"}
==> Builds finished but no artifacts were created.
2023/06/15 09:11:38 [INFO] (telemetry) Finalizing.
[1;31mBuild 'vsphere-iso.linux-sles' errored after 4 seconds 524 milliseconds: Post "https://foo/sdk": dial tcp: lookup foo on 172.29.176.1:53: server misbehaving[0m

==> Wait completed after 4 seconds 524 milliseconds

==> Some builds didn't complete successfully and had errors:
--> vsphere-iso.linux-sles: Post "https://foo/sdk": dial tcp: lookup foo on 172.29.176.1:53: server misbehaving

==> Builds finished but no artifacts were created.
2023/06/15 09:11:38 waiting for all plugin processes to complete...
2023/06/15 09:11:38 .packer.d/.packer.d/plugins/github.com/hashicorp/vsphere/packer-plugin-vsphere_v1.1.2_x5.0_linux_amd64: plugin process exited

```

</details>

I am encountering the exact same panic crash on `1.2.0` 

```
panic: interface conversion: interface is nil, not driver.Driver
```

Reverting to `1.1.2` mitigates it.

--- a/builder/vsphere/common/step_connect.go
+++ b/builder/vsphere/common/step_connect.go
@@ -68,12 +68,13 @@ func (s *StepConnect) Run(_ context.Context, state multistep.StateBag) multistep
 func (s *StepConnect) Cleanup(state multistep.StateBag) {
 	ui := state.Get("ui").(packersdk.Ui)
 	ui.Message("Closing sessions ....")
-	driver := state.Get("driver").(driver.Driver)
-	errorRestClient, errorSoapClient := driver.Cleanup()
-	if errorRestClient != nil {
-		ui.Message("Error closing rest client session: " + errorRestClient.Error())
-	}
-	if errorSoapClient != nil {
-		ui.Message("Error closing soap client session: " + errorRestClient.Error())
+	if driver, ok := state.Get("driver").(driver.Driver); ok {
+		errorRestClient, errorSoapClient := driver.Cleanup()
+		if errorRestClient != nil {
+			ui.Message("Error closing rest client session: " + errorRestClient.Error())
+		}
+		if errorSoapClient != nil {
+			ui.Message("Error closing soap client session: " + errorRestClient.Error())
+		}
 	}
 }

